### PR TITLE
fix(math): close Math gaps and follow-up regressions

### LINKED ISSUE #843: Bring Math to 100% test262 conformance (~42 fails)
## Summary

Drive the test262 **Math** failures to zero across `built-ins`, `staging`, and `intl402` — these share root causes, so a single fix typically clears the failure in more than one category.

## Why

Reaching ~100% test262 on supported categories is the objective for the conformance milestone. Baseline (main, 2026-06-26): **97.3% overall** (50,877/52,293; 0 timeouts, 0 infra; `language` and `harness` already at 100%). **Math accounts for ~42** of the 1,416 remaining failures.

## Current behavior

~42 Math tests fail. Concentrations: Math functions and edge cases (rounding, special values, sumPrecise).

## Expected behavior

Zero Math failures at the pinned suite SHA (`scripts/test262-suite-sha.txt`), with no regressions elsewhere — the PR conformance gate enforces no PASS to non-PASS transitions.

## Scope notes

- Organize the work by failing sub-area (above); they are independently fixable.
- Land the conformance gates first: harness diagnostics (#830), the crash fix (#831), the skip-list (#832).
- Every fix ships with its JS test in the same commit. Parent #820.

### PR DESCRIPTION
**Pull request by frostney**
## Summary

- Close the remaining Math test262 gaps for issue #843, including missing-argument coercion, signed-zero preservation, integer conversion, hyperbolic approximation, and `atan2` quadrant edges.
- Rework `Math.sumPrecise` to use the iterator protocol, close iterators on abrupt element errors, and accumulate finite values exactly before final double rounding.
- Fix the Date shim's TimeClip handling so the corrected `Math.trunc(-tiny) === -0` behavior does not leak `-0` into `Date.prototype.getTime()` / `valueOf()` or `Date.prototype.setTime()`.
- Reuse the resolved `Intl.DateTimeFormat` ICU formatter to keep the Temporal `formatToParts` comparison test below timeout on Linux amd64, and invalidate that cache when `Temporal.ZonedDateTime.prototype.toLocaleString` applies the ZonedDateTime time zone.
- Update focused Math, Date, Intl/Temporal regressions, plus the `Math.sumPrecise` docs, to match the implementation.

Closes #843

## Testing

- [x] Verified no regressions and confirmed the new feature or bugfix in end-to-end JavaScript/TypeScript tests
- [x] Updated documentation
- [ ] Optional: Verified no regressions and confirmed the new feature or bugfix in native Pascal tests (if AST, scope, evaluator, or value types changed)
- [ ] Optional: Verified no benchmark regressions or confirmed benchmark coverage for the change

Verification run:

- `./build.pas --clean loaderbare testrunner`
- `./build.pas testrunner`
- `./build.pas loaderbare`
- `./build/GocciaTestRunner tests/built-ins/Math --no-progress --silent`
- `./build/GocciaTestRunner tests/built-ins/Math --mode=bytecode --no-progress --silent`
- `./build/GocciaTestRunner tests/built-ins/Date --no-progress --silent`
- `./build/GocciaTestRunner tests/built-ins/Date --mode=bytecode --no-progress --silent`
- `./build/GocciaTestRunner tests/built-ins/Intl`
- `./build/GocciaTestRunner tests/built-ins/Intl --mode=bytecode`
- `./build/GocciaTestRunner tests/built-ins/Temporal/ZonedDateTime`
- `./build/GocciaTestRunner tests/built-ins/Temporal/ZonedDateTime --mode=bytecode`
- `./build/GocciaTestRunner tests/built-ins/Temporal/ZonedDateTime/prototype/toLocaleString.js`
- `./build/GocciaTestRunner tests/built-ins/Temporal/ZonedDateTime/prototype/toLocaleString.js --mode=bytecode`
- `./build/GocciaTestRunner tests --no-progress --silent`
- `./build/GocciaTestRunner tests --mode=bytecode --no-progress --silent`
- `./format.pas --check`
- `TEST262_SUITE_SHA=64ff467c0c1d60c077995bb7c5f93a9d8cc8ade1 bun scripts/run_test262_suite.ts --suite-dir /tmp/goccia-test262-843.4Q3XvM --categories built-ins,staging --filter '**/Math/**' --mode bytecode --jobs 4 --timeout-ms 20000 --output /tmp/goccia-issue-843-math-results-final-post-review.json`
- `TEST262_SUITE_SHA=64ff467c0c1d60c077995bb7c5f93a9d8cc8ade1 bun scripts/run_test262_suite.ts --suite-dir /tmp/goccia-test262-date.E5wnq8 --categories built-ins --filter 'built-ins/Date/prototype/getTime/this-value-valid-date.js' --mode bytecode --jobs 1 --timeout-ms 20000 --output /tmp/goccia-date-gettime-fixed-bytecode.json`
- `TEST262_SUITE_SHA=64ff467c0c1d60c077995bb7c5f93a9d8cc8ade1 bun scripts/run_test262_suite.ts --suite-dir /tmp/goccia-test262-date.E5wnq8 --categories built-ins --filter 'built-ins/Date/prototype/getTime/this-value-valid-date.js' --mode interpreted --jobs 1 --timeout-ms 20000 --output /tmp/goccia-date-gettime-fixed-interpreted.json`
- `TEST262_SUITE_SHA=64ff467c0c1d60c077995bb7c5f93a9d8cc8ade1 bun scripts/run_test262_suite.ts --suite-dir /tmp/goccia-test262-date.E5wnq8 --categories built-ins --filter 'built-ins/Date/prototype/valueOf/S9.4_A3_T*.js' --mode bytecode --jobs 1 --timeout-ms 20000 --output /tmp/goccia-date-valueof-fixed-bytecode.json`
- `TEST262_SUITE_SHA=64ff467c0c1d60c077995bb7c5f93a9d8cc8ade1 bun scripts/run_test262_suite.ts --suite-dir /tmp/goccia-test262-date.E5wnq8 --categories built-ins --filter 'built-ins/Date/prototype/valueOf/S9.4_A3_T*.js' --mode interpreted --jobs 1 --timeout-ms 20000 --output /tmp/goccia-date-valueof-fixed-interpreted.json`
- `TEST262_SUITE_SHA=64ff467c0c1d60c077995bb7c5f93a9d8cc8ade1 bun scripts/run_test262_suite.ts --suite-dir /tmp/goccia-test262-date.E5wnq8 --categories built-ins --filter 'built-ins/Date/TimeClip_negative_zero.js' --mode bytecode --jobs 1 --timeout-ms 20000 --output /tmp/goccia-date-timeclip-fixed-bytecode.json`
- `TEST262_SUITE_SHA=64ff467c0c1d60c077995bb7c5f93a9d8cc8ade1 bun scripts/run_test262_suite.ts --suite-dir /tmp/goccia-test262-date.E5wnq8 --categories built-ins --filter 'built-ins/Date/TimeClip_negative_zero.js' --mode interpreted --jobs 1 --timeout-ms 20000 --output /tmp/goccia-date-timeclip-fixed-interpreted.json`
- `TEST262_SUITE_SHA=64ff467c0c1d60c077995bb7c5f93a9d8cc8ade1 bun scripts/run_test262_suite.ts --suite-dir /tmp/goccia-test262-843.4Q3XvM --categories built-ins --filter '**/Date/prototype/setTime/**' --mode bytecode --jobs 4 --timeout-ms 20000 --output /tmp/goccia-pr947-settime-test262-merged.json`
- `TEST262_SUITE_SHA=64ff467c0c1d60c077995bb7c5f93a9d8cc8ade1 bun scripts/run_test262_suite.ts --suite-dir /tmp/goccia-test262-843.4Q3XvM --categories built-ins --filter '**/Date/TimeClip_negative_zero.js' --mode bytecode --jobs 4 --timeout-ms 20000 --output /tmp/goccia-pr947-timeclip-negative-zero-test262-merged.json`
- `TEST262_SUITE_SHA=64ff467c0c1d60c077995bb7c5f93a9d8cc8ade1 bun scripts/run_test262_suite.ts --suite-dir /Users/jstein/.cache/goccia-test262-64ff467c --categories intl402 --filter 'intl402/DateTimeFormat/prototype/formatToParts/compare-to-temporal.js' --mode bytecode --jobs 1 --timeout-ms 20000 --output /tmp/goccia-pr947-compare-to-temporal-bytecode-mac-cache-invalidation.json --verbose`
- `orb -m goccia-ci-amd64 bash -lc 'cd /Users/jstein/.codex/worktrees/38c8/GocciaScript && ./build.pas loaderbare && time TEST262_SUITE_SHA=64ff467c0c1d60c077995bb7c5f93a9d8cc8ade1 ~/.bun/bin/bun scripts/run_test262_suite.ts --suite-dir /Users/jstein/.cache/goccia-test262-64ff467c --categories intl402 --filter "intl402/DateTimeFormat/prototype/formatToParts/compare-to-temporal.js" --mode bytecode --jobs 1 --timeout-ms 20000 --output /tmp/goccia-pr947-compare-to-temporal-bytecode-linux-amd64-cache-invalidation.json --verbose'`

Final Math slice: 357/357 passed, 0 failed, 0 wrapper-infra.
Final Date TimeClip regression slice: 4/4 passed in bytecode and interpreted modes, 0 failed, 0 wrapper-infra.
Review Date `setTime` slice: 11/11 passed in bytecode mode, 0 failed, 0 wrapper-infra.
Final Intl/Temporal focused suites: Intl 277/277 passed in both modes; Temporal/ZonedDateTime 66/66 passed in both modes.
Final `compare-to-temporal.js` timeout slice: passed on macOS in 3.1s and on OrbStack Linux amd64 in 4.6s harness / 5.2s real.

### CODE REVIEW

File: source/units/Goccia.Shims.pas [hunk 1]
```diff
@@ -484,7 +484,8 @@ implementation
         '};'#10 +
         'const __GocciaDateTimeClip = (epochMilliseconds: any): number => {'#10 +
         '  const t: number = Math.trunc(Number(epochMilliseconds));'#10 +
-        '  return Number.isFinite(t) && Math.abs(t) <= dateTimeClipLimit ? t : NaN;'#10 +
+        '  if (!Number.isFinite(t) || Math.abs(t) > dateTimeClipLimit) return NaN;'#10 +
+        '  return t === 0 ? 0 : t;'#10 +
         '};'#10 +
```
[the remark this answers is not in the document]
Comment (frostney) [reply source/units/Goccia.Shims.pas:489]: Fixed in 02e0c1b2. Date.#clip now delegates to the shared __GocciaDateTimeClip helper, so setTime uses the same +0 zero canonicalization as the constructor path. Added a setTime regression for -0 and tiny negative values, with Date suite and focused test262 setTime coverage passing.
